### PR TITLE
[OSCD] rpcping lolbin

### PR DESCRIPTION
**Pull request by Vasilisa-L**
Detects using Rpcping.exe to send a RPC test connection to the target server (-s) and force the NTLM hash to be sent in the process.
#1014

### CODE REVIEW

**Comment by aw350m33d** [comment]
The parameter `-t` seems to have no dash. 
And, perhaps, for this parameter, both options for writing the parameter are possible. 

Can you check it out?

**Comment by Vasilisa-L** [comment]
@aw350m33d 
I intentionally omitted both "-" and "/" before key ` -t` not to create multiple conditions.
It's implied that string "ncacn_np" follows key `-t`  (or `/t`) and ususally can't follow anything else

Or is it better to make rule more concrete and specify both variants  `-t`  and `/t`?

**Comment by aw350m33d** [comment]
I just do not understand why you distinguish between cases with parameter `t` and with parameter `u`. It seems to be the same situation, isn't it?

**Comment by Vasilisa-L** [comment]
Nein
In case of parameter `u` i need to know that this key is used and it doesn't matter which value is used
In the second case it's more important to detect value `ncacn_np` than `t` parameter usage

**Comment by Vasilisa-L** [comment]
@aw350m33d 
Ok, i got your point and fixed the rule
Is it okay now?

And i have no idea why tests do not pass :no_mouth:

**Comment by aw350m33d** [comment]
I think this is a problem with the tests themselves. The rule looks correct. 

Thank you very much for your hard work! I believe that the rule can be merged. @yugoslavskiy

File: rules/windows/process_creation/win_susp_rpcping.yml [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+    remote_server:
+        CommandLine|contains: 
+            - '-s'
+            - '/s'
+    ntlm_auth:
+        CommandLine|contains: 
+            - '-u NTLM'
+            - '/u NTLM'
+            - '-t ncacn_np'
+            - '/t ncacn_np'
```
Comment (yugoslavskiy) [inline rules/windows/process_creation/win_susp_rpcping.yml:29]: could you please refactor this part not using spaces? here is an option:

```
    ntlm_auth:
        - CommandLine|contains|all: 
            - '-u'
            - 'NTLM'
        - CommandLine|contains|all: 
            - '/u'
            - 'NTLM'
        - CommandLine|contains|all: 
            - '-t'
            - 'ncacn_np'
        - CommandLine|contains|all: 
            - '/t'
            - 'ncacn_np'
```

File: rules/windows/process_creation/win_susp_rpcping.yml [hunk 2]
```diff
@@ -0,0 +1,33 @@
+title: Capture Credentials with Rpcping.exe
+id: 93671f99-04eb-4ab4-a161-70d446a84003
+description: Detects using Rpcping.exe to send a RPC test connection to the target server (-s) and force the NTLM hash to be sent in the process.
+status: experimental
+references:
+    - https://twitter.com/vysecurity/status/974806438316072960
+    - https://twitter.com/vysecurity/status/873181705024266241
+    - https://docs.microsoft.com/en-us/previous-versions/windows/it-pro/windows-server-2012-r2-and-2012/hh875578(v=ws.11)
```
Comment (yugoslavskiy) [inline rules/windows/process_creation/win_susp_rpcping.yml:8]: Could you please add a link to the original LOLBAS page?

File: rules/windows/process_creation/win_susp_rpcping.yml (hunk 1 above)
Comment (Vasilisa-L) [reply rules/windows/process_creation/win_susp_rpcping.yml]: Great idea

**Comment by yugoslavskiy** [comment]
Well done, @Vasilisa-L! Thank you very much!

**Review by yugoslavskiy — APPROVED** [review]
@thomaspatzke @Neo23x0 guys, I believe that this PR could be merged (:
please let us know what do you think about it if there is anything you would like to change.

**Pull request merged**